react-components/src/query/useSearchMappedEquipmentAssetMappings.tsx

```diff
@@ -119,6 +119,58 @@ export const useAllMappedEquipmentAssetMappings = (
   });
 };
 
+export const useMappingsForAssetIds = (
+  models: AddModelOptions[],
+  assetIds: number[]
+): UseInfiniteQueryResult<InfiniteData<ModelMappingsWithAssets[]>, Error> => {
+  const sdk = useSDK();
+
+  return useInfiniteQuery({
+    queryKey: [
+      'reveal',
+      'react-components',
+      'mappings-for-asset-ids',
+      ...models.map((model) => [model.modelId, model.revisionId]),
+      ...assetIds
+    ],
+    queryFn: async ({ pageParam = models.map((model) => ({ cursor: 'start', model })) }) => {
+      const currentPagesOfAssetMappingsPromises = models.map(async (model) => {
+        const nextCursors = pageParam as Array<{
+          cursor: string | 'start' | undefined;
+          model: AddModelOptions;
+        }>;
+
+        const nextCursor = nextCursors.find(
+          (nextCursor) =>
+            nextCursor.model.modelId === model.modelId &&
+            nextCursor.model.revisionId === model.revisionId
```

Comment:
I think you may get away with just a `nextCursor.model === model` here

Reply:
Making sure models are same, I prefer this way!